metric/emitter/prometheus.go

```diff
@@ -421,3 +435,33 @@ func (emitter *PrometheusEmitter) databaseMetrics(logger lager.Logger, event met
 	}
 
 }
+
+// updateLastSeen tracks for each worker when it last received a metric event.
+func (emitter *PrometheusEmitter) updateLastSeen(event metric.Event) {
+	emitter.mu.Lock()
+	defer emitter.mu.Unlock()
+	if worker, exists := event.Attributes["worker"]; exists {
+		emitter.workerLastSeen[worker] = time.Now()
+	}
+}
+
+//periodically remove stale metrics for workers
+func (emitter *PrometheusEmitter) periodicMetricGC() {
+	for {
+		emitter.mu.Lock()
+		now := time.Now()
+		for worker, lastSeen := range emitter.workerLastSeen {
+			if now.Sub(lastSeen) > 5*time.Minute {
+				//this is a little stupid but we don't know the platform here,
+				//so we try all known platoforms
+				for _, platform := range []string{"linux", "windows", "darwin"} {
+					emitter.workerContainers.DeleteLabelValues(worker, platform)
```

Comment:
Does this need the platform? Shouldn't the worker name be enough? Those are globally unique within Concourse's database.

Reply:
The problem is that `DeleteLabelValues` only removes timeseries that have an exact match ok the label set. 

See https://godoc.org/github.com/prometheus/client_golang/prometheus#GaugeVec.DeleteLabelValues

Reply:
Gotcha, so it runs against all platforms just to cover all the bases. Wish there was a better way to do that, but it makes sense.

Unfortunately there are conflicts now so I'll hold off for a rebase, but I'll approve the changes for now.